scripts/setup/nxp/update_nxp_sdk.py

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+CHIP_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../..'))
+
+# Array listing all SDK and default targeted path for NXP SDK storage
+ALL_PLATFORM_SDK = [
+    {'plat_name': 'k32w0', 'plat_sdk_manifest_path': 'third_party/nxp/nxp_matter_support/github_sdk/k32w0'},
+]
+
+all_platform_sdk_list = list(map(lambda plat: plat["plat_name"], ALL_PLATFORM_SDK))
+
+
+def clean_sdk_local_changes(sdk_target_location):
+    print("SDK will be cleaned all local modification(s) will be lost")
+    # Cleaning all local modifications
+    git_clean_command = "git reset --hard && git clean -xdf"
+    command = ['west', 'forall', '-c', git_clean_command, '-a']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+
+
+def init_platform_sdk_version(sdk_target_location, force):
+    print("Init SDK in: " + sdk_target_location)
+    if not force and os.path.exists(os.path.join(sdk_target_location, '.west')):
+        print("Error SDK is already initialized, use --force to force init")
+        print("WARNING -- All local SDK modification(s) will be lost")
+        sys.exit(1)
+    command = ['rm', '-rf', '.west']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+
+    command = ['west', 'init', '-l', 'manifest', '--mf', 'west.yml']
+    subprocess.run(command, cwd=sdk_target_location, check=True)
+    update_platform_sdk_version(sdk_target_location, force)
+
+
+def update_platform_sdk_version(sdk_target_location, force):
+    print("Update SDK in " + sdk_target_location)
+    if not os.path.exists(os.path.join(sdk_target_location, '.west')):
+        print("--update-only error SDK is not initialized")
+        sys.exit(1)
+    command = ['west', 'update', '--fetch', 'smart']
+    try:
+        subprocess.run(command, cwd=sdk_target_location, check=True)
+    except (RuntimeError, subprocess.CalledProcessError) as exception:
+        if force:
+            # In case of force update option and in case of update failure:
+            # 1. try to clean all local modications if any
+            # 2. Retry the west update command. It should be successfull now as all local modifications have been cleaned
+            clean_sdk_local_changes(sdk_target_location)
+            subprocess.run(command, cwd=sdk_target_location, check=True)
+        else:
+            print(exception)
```

Comment:
consider using python logging throughout. Then you can use `logging.exception` to report the exception with a full stack trace and info. otherwise probably traceback would be useful.

Reply:
Yes agree, I switched to python logging in the last commit.